### PR TITLE
feat: add test script for cli server

### LINKED ISSUE #38: add a test script for kubeagi_cli

### PR DESCRIPTION
**Pull request by Lanture1064**
#### What type of PR is this?

/kind feature

#### Which issue(s) this PR fixes

Fixes #38

### CODE REVIEW

File: .github/workflows/cli_test.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        - name: Set up GCC
+          uses: egor-tensin/setup-gcc@v1
+          with:
+            version: latest
+            platform: x64
+        - name: Build core library cli image
+          id: push-worker
+          uses: docker/build-push-action@v5
+          with:
+            context: .
```
Comment (Abirdcfly) [inline .github/workflows/cli_test.yaml:38]: add `load: true` ref :https://docs.docker.com/build/ci/github-actions/export-docker/

File: .github/workflows/cli_test.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            build-args: |
+              PACKAGE_REGISTRY=${{ env.PACKAGE_REGISTRY }}
+              PYTHON_INDEX_URL=${{ env.PYTHON_INDEX_URL }}
+        - name: Run Docker Container
+          run: |
+            docker run -d --name temp-container test-image
+            docker ps
+        - name: Run test script
+          run: |
+            sh -C tests/example_test.sh
```
Comment (Abirdcfly) [inline .github/workflows/cli_test.yaml:52]: It looks like the top part is similar to https://github.com/kubeagi/core-library/blob/main/.github/workflows/image_build_test.yml, so I guess you could just modify that file?

Comment (Lanture1064) [reply .github/workflows/cli_test.yaml:53]: I simply cp those into the test action yaml for sure. Could we merge all tests together or leave it here as we might need other tests that would be better executed seperately?

File: .github/workflows/cli_test.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    steps:
+        - uses: actions/checkout@v4
+          with:
+            fetch-depth: 0
+        - name: Maximize build space
```
Comment (Abirdcfly) [inline .github/workflows/cli_test.yaml:20]: `Maximize build space` should be before `actions/checkout@v4`

File: tests/example_test.sh [hunk 4]
```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
+
+# Install curl
+apt-get install -y curl
```
[the remark this answers is not in the document]
Comment (Lanture1064) [reply tests/example_test.sh:4]: The shell is executed inside the running container, which doesn't have curl

**Pull request merged**